symengine/polys.h

```diff
@@ -0,0 +1,98 @@
+/**
+ *  \file polys.h
+ *  Class for sparse Polynomial: Polynomial
+ *
+ **/
+#ifndef SYMENGINE_POLYS_H
+#define SYMENGINE_POLYS_H
+
+#include <symengine/basic.h>
+#include <symengine/dict.h>
+#include <symengine/symbol.h>
+#include <unordered_set>
+
+namespace SymEngine {
+
+//! Polynomial Class
+class Polynomial : public Basic{
+public:
+    //! `vars_` : Variables of the Polynomial
+    //! `polys_set_` : holds the Polynomial
+    vec_symbol vars_;
+    #ifdef HAVE_SYMENGINE_PIRANHA
+    hash_set polys_set_;
+    #else
+    unordered_set polys_set_;
```

Comment:
Here, you can do
```
#ifdef HAVE_SYMENGINE_PIRANHA
   typedef hash_set polys_set;
#else
   typedef unordered_set polys_set;
#endif
   polys_set polys_set_;
```
And use `polys_set` as the container used everywhere. Avoids a lot of #define statements